templates/todo/projects/python-mongo/.repo/terraform/infra/main.tf

```diff
@@ -0,0 +1,306 @@
+  resource_type = "azurerm_key_vault"
+  random_length = 0
+  clean_input   = true
+}
+
+resource "azurerm_key_vault" "kv" {
+  name                     = azurecaf_name.kv_name.result
+  location                 = azurerm_resource_group.rg.location
+  resource_group_name      = azurerm_resource_group.rg.name
+  tenant_id                = data.azurerm_client_config.current.tenant_id
+  purge_protection_enabled = false
+  sku_name                 = "standard"
+
+  tags = local.tags
+}
+
+resource "azurerm_key_vault_access_policy" "app" {
+  key_vault_id = azurerm_key_vault.kv.id
+  tenant_id    = data.azurerm_client_config.current.tenant_id
+  object_id    = azurerm_linux_web_app.api.identity.0.principal_id
+
+  secret_permissions = [
+    "Get",
+    "Set",
+    "List",
+    "Delete",
+  ]
+}
+
+resource "azurerm_key_vault_access_policy" "user" {
+  count        = var.principalId == "" ? 0 : 1
+  key_vault_id = azurerm_key_vault.kv.id
+  tenant_id    = data.azurerm_client_config.current.tenant_id
+  object_id    = var.principalId
+
+  secret_permissions = [
+    "Get",
+    "Set",
+    "List",
+    "Delete",
+
+  ]
+}
+
+resource "azurerm_key_vault_secret" "dbconnection" {
+  name         = "AZURE-COSMOS-CONNECTION-STRING"
+  value        = azurerm_cosmosdb_account.db.connection_strings[0]
+  key_vault_id = azurerm_key_vault.kv.id
+
+}
+# ------------------------------------------------------------------------------------------------------
+# Deploy app service plan
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "plan_name" {
+  name          = random_string.resource_token.result
+  resource_type = "azurerm_app_service_plan"
+  random_length = 0
+  clean_input   = true
+}
+resource "azurerm_service_plan" "plan" {
+  name                = azurecaf_name.plan_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+
+  tags = local.tags
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Deploy app service web app
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "web_name" {
+  name          = "web-${random_string.resource_token.result}"
+  resource_type = "azurerm_app_service"
+  random_length = 0
+  clean_input   = true
+}
+resource "azurerm_linux_web_app" "web" {
+  name                = azurecaf_name.web_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  service_plan_id     = azurerm_service_plan.plan.id
+  https_only          = true
+  tags                = merge(local.tags, { azd-service-name : "web" })
+
+  site_config {
+    #linux_fx_version = "NODE|16-lts"

```

Comment:
Should we just delete this (I'm guessing it's here because we have it in the ARM version, but I see the provider doesn't have an `linux_fx_version` property).  I wonder if we were intended to set a property in the `application_stack` block to set `node_version` to `16-lts`?

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_web_app#node_version

Reply:
Yeah, look like `node_version` is what we want here.
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_web_app#node_version

Reply:
added the config